apps/web/src/app/tracks/_components/enroll-button.tsx

```diff
@@ -0,0 +1,20 @@
+'use client';
+
+import { Button } from '@repo/ui';
+
+interface EnrollButtonProps {
+  action: () => Promise<void>;
+  text: string;
+}
+
+export function EnrollButton({ action, text }: EnrollButtonProps) {
+  return (
+    <Button
+      onClick={async () => {
+        await action();
```

Comment:
error handling?

Reply:
oh yep, i actually had error handling in the `track-details.tsx` because i wanted custom messages. Although u give me an idea, wait i'll test it out. I was stupid & didn't think of this before. 💀

Reply:
RESOLVEDEDD!!!